src/scf/nbands_algorithm.jl

```diff
@@ -68,6 +68,8 @@ function determine_n_bands(bands::AdaptiveBands, occupation::AbstractVector,
     # TODO Could return different bands per k-Points
 
     # Determine number of bands to be actually converged
+    # Bring occupation on the CPU, or findlast will fail
+    occupation = [to_cpu(occk) for occk in occupation]
```

Comment:
When this is merged, can you open an issue @GVigne to indicate that we should reduce the transfer to CPU of the occupation array.